src/Runner/Parallel/ProcessFactory.php

```diff
@@ -61,6 +61,8 @@ public function getCommandArgs(int $serverPort, ProcessIdentifier $identifier, R
             throw new ParallelisationException('Cannot find PHP executable.');
         }
 
+        $phpBinary = str_replace(' ', '\ ', $phpBinary);
```

Comment:
please, add a comment above with description why we do this.
ideally, a test case to cover this case.

otherwise, it will be very easy to re-introduce this issue.

Reply:
I am not sure if this is the valid approach. Wouldn't it fix the problem if we do `escapeshellarg($phpBinary)` in line 77, where `$commandArgs` are created? This should not only escape space in the PHP binary path on Herd, but also would handle any other characters and ensure the PHP binary path is treated as a single argument.

Reply:
That's an even better approach! I have tested that and works perfectly and doesn't require an upstream change.

Reply:
@Wirone any reason this ProcessFactory builds an array that is imploded later to generate a command line (that requires proper escaping which is easy to forget) instead of using the array-based signature of the Process constructor, where symfony/process takes care of any necessary escaping for you ?